climada/hazard/base.py

```diff
@@ -1396,47 +1394,99 @@ def _read_att_excel(self, file_name, var_names):
                              f'{dfr.shape[0]} != {self.centroids.size}')
 
         self.intensity = sparse.csr_matrix(dfr.values[:, 1:num_events + 1].transpose())
-        self.fraction = sparse.csr_matrix(np.ones(self.intensity.shape,
-                                                  dtype=np.float))
+        self.fraction = sparse.csr_matrix(np.ones(self.intensity.shape, dtype=float))
 
-    def append(self, hazard):
-        """Append the events and centroids in hazard.
+    def append(self, *haz_list):
+        """Append the events and centroids to this hazard object.
 
-        Hazard must be of same type as self.
-        Centroids of all hazards must have the same CRS.
+        All of the given hazards must be of the same type as self.
+        The centroids of all hazards must have the same CRS.
 
-        Tthe centroids of both hazards are combined together.
+        The centroids of all hazards are combined together.
         All raster centroids are converted to points and raster data
         is discarded.
 
-        Note: centroids of self is modified in place and raster information
+        Note: `self.centroids` is modified in place and centroid raster information (meta)
         is destroyed.
 
-        It is recommended to use the static method `hazard.concatenate_hazard`
-        which concatenates a list of hazards into a new object.
-
         Parameters
         ----------
-        hazard: climada.hazard.Hazard object
-            Hazard instance to append to self
+        haz_list : one or more climada.hazard.Hazard objects
+            Hazard instances to append to self
 
         Raises
         ------
         TypeError
 
         See Also
         --------
-        hazard.concat: concatenate 2 or more hazards
+        Hazard.concat: concatenate 2 or more hazards
         """
-        hazard._check_events()
-        if type(self) != type(hazard):
-            raise TypeError(f"hazard is of class {type(self)} which is "
-                            f" different from {type(hazard)}")
+        if len(haz_list) == 0:
+            return
+        haz_list = (self,) + haz_list
+        haz_list_nonempty = [haz for haz in haz_list if haz.size > 0]
```

Comment:
For the actual concatenation of array/list type attributes later, only non-empty Hazard objects are relevant, so we keep track of those separately.